File: symfony-migration/src/Entity/Offre.php
<?php

namespace App\Entity;

use App\Repository\OffreRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: OffreRepository::class)]
#[ORM\Table(name: 'offre')]
class Offre
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'idOffre')]
    private ?int $id = null;

    #[ORM\Column(name: 'nomOffre', length: 255)]
    private string $nomOffre = '';

    #[ORM\Column(name: 'type', enumType: TypeOffre::class)]
    private TypeOffre $type = TypeOffre::CDI;

    #[ORM\Column(name: 'competences', type: 'text')]
    private string $competences = '';

    #[ORM\Column(name: 'salaire')]
    private int $salaire = 0;

    public function getId(): ?int { return $this->id; }
    public function getNomOffre(): string { return $this->nomOffre; }
    public function setNomOffre(string $nomOffre): self { $this->nomOffre = $nomOffre; return $this; }
    public function getType(): TypeOffre { return $this->type; }
    public function setType(TypeOffre $type): self { $this->type = $type; return $this; }
    public function getCompetences(): string { return $this->competences; }
    public function setCompetences(string $competences): self { $this->competences = $competences; return $this; }
    public function getSalaire(): int { return $this->salaire; }
    public function setSalaire(int $salaire): self { $this->salaire = $salaire; return $this; }
}

File: src/main/java/models/Offre.java
package models;

public class Offre {
    private int idOffre;
    private String nomOffre;
    private TypeOffre type;
    private String competences;
    private int salaire;

    public Offre(int idOffre, String nomOffre, TypeOffre  type, String competences, int salaire) {
        this.idOffre = idOffre;
        this.nomOffre = nomOffre;
        this.type = type;
        this.competences = competences;
        this.salaire = salaire;
    }

    public Offre(String nomOffre, TypeOffre  type, String competences, int salaire) {
        this.nomOffre = nomOffre;
        this.type = type;
        this.competences = competences;
        this.salaire = salaire;
    }

    public Offre() {
    }

    public int getIdOffre() {
        return idOffre;
    }

    public void setIdOffre(int idOffre) {
        this.idOffre = idOffre;
    }

    public String getNomOffre() {
        return nomOffre;
    }

    public void setNomOffre(String nomOffre) {
        this.nomOffre = nomOffre;
    }

    public TypeOffre  getType() {
        return type;
    }

    public void setType(TypeOffre  type) {
        this.type = type;
    }

    public String getCompetences() {
        return competences;
    }

    public void setCompetences(String competences) {
        this.competences = competences;
    }

    public int getSalaire() {
        return salaire;
    }

    public void setSalaire(int salaire) {
        this.salaire = salaire;
    }

    @Override
    public String toString() {
        return "Offre{" +
                "idOffre=" + idOffre +
                ", nomOffre='" + nomOffre + '\'' +
                ", type='" + type + '\'' +
                ", competences='" + competences + '\'' +
                ", salaire=" + salaire +
                '}';
    }
}
